test/e2e/e2e.go

```diff
@@ -713,33 +683,484 @@ var _ = Describe("validation", func() {
 						},
 					},
 				}
-				Eventually(func() bool {
-					err := testclient.Client.Delete(context.Background(), addresspool)
-					return errors.IsNotFound(err)
-				}, timeout, interval).Should(BeTrue(), "Failed to delete AddressPool custom resource")
+				err := testclient.Client.Delete(context.Background(), addresspool)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			// Make sure Configmap is deleted at the end of this test
+			By("Checking ConfigMap is deleted at the end of the test")
+			Eventually(func() bool {
+				_, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+				return errors.IsNotFound(err)
+			}, timeout, interval).Should(BeTrue())
+		})
+	})
+
+	Context("Creating BGP Peer", func() {
```

Comment:
What are we gaining from this context and the following one? Do they differ from those in the controller tests?

Reply:
two context one was just basic peer and another with peer and address pool they are covered by ut as well but I wanted to have the same coverage on real cluster too just to be sure since both env are different ?

Reply:
After talking a bit we didn't get to an agreement so we wanted to hear what you have to say @sabinaaledort @schseba:
The thing is that the tests here (e2e) do practically the same as those added to the controllers' integration tests. I feel like those in the e2e that do not test any behavior that actually requires a live cluster should not be here - because of code duplication/ not much benefit is added / ... 
The benefits that are added are something like ensuring the rbacs work/the ability to test against a live cluster (not necessarily one created for the test). We could keep it this way / keep them to a minimum in e2e suite / remove them from e2e / other suggestions ?

Reply:
i think we should avoid code duplication so maybe better add them only to the integration tests as there is no special cluster setup needed here.

Reply:
ok fine, I just kept basic BGP peer testing here and let the full testing with addresspool test handled by UT